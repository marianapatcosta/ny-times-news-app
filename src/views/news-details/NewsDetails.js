import { defineComponent } from "vue";
import { Image } from "@/components";
import { formatDate } from "@/utils/utils";

export default defineComponent({
  name: "NewsDetails",
  components: {
    Image,
  },
  computed: {
    newsInfo() {
      return this.$store.getters.newsInfo;
    },
  },
  data() {
    return {
      formatDate,
    };
  },
});
