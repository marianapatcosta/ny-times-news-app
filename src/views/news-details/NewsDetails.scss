$space: 0.75rem;

.news-details {
  max-width: 45rem;
  margin: 0 auto;
  padding-bottom: 2rem;

  &__image {
    width: 100%;
    margin: $space 0;
  }

  &__section-name {
    font-weight: 700;
  }

  &__title {
    margin: $space 0;
  }

  &__date {
    font-style: italic;
  }

  &__abstract,
  &__source {
    margin: $space 0;
  }

  &__source {
    font-style: italic;
  }

  &__read-more {
    text-align: right;
    a {
      background-color: $color-secondary;
      box-shadow: 0 0.05rem 0.3rem rgba($color-black, 0.5);
      color: $color-primary;
      display: inline-block;
      padding: 0.5rem 1rem;
      margin-top: 1rem;
      margin-left: auto;
      text-decoration: none;
    }
  }

  &__no-data {
    font-size: 130%;
    margin-top: 10rem;
    text-align: center;
  }
}
