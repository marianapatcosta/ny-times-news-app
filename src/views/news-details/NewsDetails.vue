<template>
  <article v-if="newsInfo" class="news-details" data-cy="news-details">
    <header>
      <div class="news-details__section-name">
        {{ newsInfo.sectionName }}
      </div>
      <h1 class="news-details__title">{{ newsInfo.headline }}</h1>
      <time class="news-details__date" :datetime="newsInfo.pubDate">
        {{ formatDate(newsInfo.pubDate) }}
      </time>
      <div class="news-details__abstract">{{ newsInfo.abstract }}</div>
      <Image
        class="news-details__image"
        v-if="newsInfo.imageUrl"
        :src="`https://www.nytimes.com/${newsInfo.imageUrl}`"
        :alt="newsInfo.headline"
      />
    </header>
    <div class="news-details__content">
      <div class="news-details__details">{{ newsInfo.leadParagraph }}</div>
      <div class="news-details__source" v-if="newsInfo.source">
        Source: {{ newsInfo.source }}
      </div>
      <div class="news-details__read-more">
        <a
          data-cy="read-more-button"
          :href="newsInfo.webUrl"
          target="_blank"
          rel="nofollow noopener noreferrer"
          >read more
        </a>
      </div>
    </div>
  </article>
  <p v-else class="news-details__no-data">There is no news data to display.</p>
</template>
<script src="./NewsDetails.js"></script>
<style scoped lang="scss" src="./NewsDetails.scss"></style>
