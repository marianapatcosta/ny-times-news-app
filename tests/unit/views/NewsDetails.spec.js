import { createStore } from "vuex";
import { mount } from "@vue/test-utils";
import NewsDetails from "@/views/news-details/NewsDetails.vue";
import { newsInfoMock } from "../../mocks";

describe("NewsDetails.vue", () => {
  let wrapper;
  let store;
  const mockedNewsInfoGetter = jest.fn();
  mockedNewsInfoGetter.mockReturnValue(newsInfoMock);
  beforeEach(() => {
    const getters = {
      newsInfo: mockedNewsInfoGetter,
    };

    store = createStore({
      getters,
    });

    wrapper = mount(NewsDetails, {
      global: {
        mocks: {
          $store: store,
        },
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render without errors and match snapshot renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should call newsInfo getter to get newsInfo from the store when rendered", () => {
    expect(wrapper.vm.newsInfo).toBeTruthy();
    expect(mockedNewsInfoGetter).toHaveBeenCalled();
  });
});
