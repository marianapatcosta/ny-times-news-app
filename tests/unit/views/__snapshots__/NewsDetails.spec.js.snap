// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsDetails.vue should render without errors and match snapshot renders correctly 1`] = `
<article
  class="news-details"
  data-cy="news-details"
>
  <header>
    <div
      class="news-details__section-name"
    >
      T Magazine
    </div>
    <h1
      class="news-details__title"
    >
      The headline
    </h1>
    <time
      class="news-details__date"
      datetime="2021-10-15T11:00:12+0000"
    >
      October 15, 2021
    </time>
    <div
      class="news-details__abstract"
    >
      Lorem ipsum dolor sit amet. Non fugiat mollitia est nihil fugiat ea amet exercitationem. Et similique minus qui voluptas galisum et culpa tempora ea reprehenderit repudiandae.
    </div>
    <div
      class="image__wrapper news-details__image"
    >
      <div
        class="image__placeholder"
      />
      <img
        alt="The headline"
        class="image"
        loading="lazy"
        src="https://www.nytimes.com/images/2021/10/15/t-magazine/15tmag-harlemymca-slide-AEU3-copy/15tmag-harlemymca-slide-AEU3-copy-articleLarge.jpg"
        style="display: none;"
      />
    </div>
  </header>
  <div
    class="news-details__content"
  >
    <div
      class="news-details__details"
    >
      Lorem ipsum dolor sit amet. Sed voluptatem architecto ut culpa soluta nam autem tempore? A repudiandae dolorem a rerum praesentium hic eveniet quia est dolores magnam ut odit saepe hic autem voluptatibus. Ab sunt esse sed nihil reiciendis non sapiente iusto est doloremque dolorem. Ex debitis cum nulla officiis ea repellendus nemo.
    </div>
    <div
      class="news-details__source"
    >
       Source: The New York Times
    </div>
    <div
      class="news-details__read-more"
    >
      <a
        data-cy="read-more-button"
        href="https://www.nytimes.com/2021/10/15/t-magazine/harlem-ymca-black-culture.html"
        rel="nofollow noopener noreferrer"
        target="_blank"
      >
        read more 
      </a>
    </div>
  </div>
</article>
`;
